src/server/info_collector.cpp

```diff
@@ -285,7 +289,17 @@ hotspot_calculator *info_collector::get_hotspot_calculator(const std::string &ap
     if (iter != _hotspot_calculator_store.end()) {
         return iter->second;
     }
-    hotspot_calculator *calculator_address = new hotspot_calculator(app_name, partition_num);
+    std::unique_ptr<hotspot_policy> policy;
+    if (_hotspot_detect_algorithm == "hotspot_algo_qps_variance") {
+        policy.reset(new hotspot_algo_qps_variance());
+    } else if (_hotspot_detect_algorithm == "hotspot_algo_qps_skew") {
+        policy.reset(new hotspot_algo_qps_skew());
+    } else {
+        ddebug("no such hotspot detect algorithm");
+        policy.release();
```

Comment:
Return `nullptr` here, and you should check whether it work when `get_hotspot_calculator` return `nullptr`, may add some protective code.